dthm4kaiako/events/views.py

```diff
@@ -124,3 +206,1524 @@ class LocationDetailView(RedirectToCosmeticURLMixin, generic.DetailView):
+        manage_registration_form_details_form = ManageEventRegistrationFormDetailsForm(
+            request.POST,
+            instance=registration_form
+            )
+        if manage_registration_form_details_form.is_valid():
+
+            updated_open_datetime = manage_registration_form_details_form.cleaned_data['open_datetime']
+            updated_close_datetime = manage_registration_form_details_form.cleaned_data['close_datetime']
+            updated_terms_and_conditions = manage_registration_form_details_form.cleaned_data['terms_and_conditions']
+
+            RegistrationForm.objects.filter(event_id=registration_form.pk).update(
+                open_datetime=updated_open_datetime,
+                close_datetime=updated_close_datetime,
+                terms_and_conditions=updated_terms_and_conditions,
+            )
+            registration_form.save()
+            messages.success(request, f'You have updated the event registration form details of {event.name}')
+            return HttpResponseRedirect(reverse("events:event_management", kwargs={'pk': event.pk}))
+        else:
+            messages.warning(
+                request,
+                f'Please resolve the invalid fields to update the registration form details of \"{event.name}\".'
+            )
+
+    context['manage_registration_form_details_form'] = manage_registration_form_details_form
+    context['event'] = event
+    context['registration_form_pk'] = registration_form.pk
+    context['event_pk'] = event.pk
+
+    return render(request, 'events/event_management.html', context)
+
+
+def convert_string_list_to_one_string(listToConvert):
+    """Convert list to string.
+
+    Returns:
+        A string of values separated by &'s.
+    """
+    if len(listToConvert) == 1:
+        return listToConvert[0]
+    else:
+        newBigString = ""
+        for i in range(0, len(listToConvert)):
+            currentString = listToConvert[i]
+            if i == len(listToConvert) - 1:
+                newBigString += currentString
+            else:
+                newBigString += currentString + " & "
+        return newBigString
+
+
+# TODO: fix UI bug where the validation error message only disappears
+# if go back out and back to events management hub page
+# TODO: add staff and admin permissions
+@login_required
+def generate_event_csv_view(request):
+    """Generate a custom CSV of events' data."""
+    event = Event.objects.get(pk=request.event.pk)
+
+    if not can_view_event_management_content(request, event):
+        messages.warning(
+            request,
+            NON_EVENT_STAFF_ACCESS_MESSAGE
+        )
+        return HttpResponseRedirect(reverse("events:events_management"))
+
+    if request.method == 'POST':
+        builderFormForEventsCSV = BuilderFormForEventsCSV(request.POST)
+        if builderFormForEventsCSV.is_valid():
+
+            file_name = builderFormForEventsCSV.cleaned_data['file_name']
+
+            first_wrote_titles = []
```

Comment:
Is this supposed to be `first_row_titles`?

Reply:
Yes, thanks for catching that one!